subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/TestProjectGeneratorRunner.groovy

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.generator
+
+class TestProjectGeneratorRunner {
+
+    static void main(String[] args) {
+        def projectName = args[0]
+        def outputDir = new File(args[1])
+
+        JavaTestProject project = JavaTestProject.values().find { it.projectName == projectName }
```

Comment:
This is fine for now, but if we want to make this reusable then we can't have a hardcoded enum, but actually need to take the configuration from the command line or a file.